soc/arm/cypress/Kconfig

```diff
@@ -29,6 +29,7 @@ config SOC_PSOC6_M0_ENABLES_M4
 
 config SOC_FAMILY_PSOC6
 	bool
+	select USE_INFINEON_WDT
```

Comment:
this likely needs to be selected by a wdt driver. note that HAL should not be "smart" by enabling devices on its own, unless there's a reason.

Reply:
it required by SystemInit (https://github.com/zephyrproject-rtos/hal_infineon/blob/4af06965f57ba1e7d170e6a97d24c33785543a8c/mtb-pdl-cat1/devices/COMPONENT_CAT1A/templates/COMPONENT_MTB/COMPONENT_CM4/system_psoc6_cm4.c#L157), where we Unlock and disable WDT during PSoC 6 init.

Reply:
updated